### PR TITLE
Validate path to Conan executable

### LINKED ISSUE #117: Validate Conan executable
If the path to Conan is not correct, there is an "AccessDenied" error (make it a meaninful error). We need to validate the path from the very beginning and maybe print the result of a `conan --version` in the config window.

### PR DESCRIPTION
**Pull request by SSE4**
closes #117

### CODE REVIEW

**Comment by jgsogo** [comment]
Rebase with `dev` and solve conflicts, please

File: Conan.VisualStudio/ConanOptionsPage.cs [hunk 1]
```diff
@@ -12,14 +15,46 @@ public class ConanOptionsPage : DialogPage
...
+        }
 
         [Category("Conan")]
         [DisplayName("Conan executable")]
         [Description(@"Path to the Conan executable file, like C:\Python27\Scripts\conan.exe")]
         public string ConanExecutablePath
         {
             get => _conanExecutablePath ?? (_conanExecutablePath = ConanPathHelper.DetermineConanPathFromEnvironment());
-            set => _conanExecutablePath = value;
+            set => _conanExecutablePath = ValidateConanExecutable(value) ? value : _conanExecutablePath;
```
Comment (jgsogo) [inline Conan.VisualStudio/ConanOptionsPage.cs]: I think we should validate the path returned by `ConanPathHelper.DetermineConanPathFromEnvironment()` too. I'm not sure if this is the place to do this, or it is better to validate it inside the function before returning (return `null` if invalid)

Comment (SSE4) [reply Conan.VisualStudio/ConanOptionsPage.cs]: I think it should be done inside function then

Comment (jgsogo) [reply Conan.VisualStudio/ConanOptionsPage.cs]: Sure, please add that check

Comment (SSE4) [reply Conan.VisualStudio/ConanOptionsPage.cs]: added

**Comment by SSE4** [comment]
@jgsogo  rebased

File: Conan.VisualStudio.Core/ConanPathHelper.cs [hunk 2]
```diff
@@ -23,6 +25,37 @@ public static string GetRelativePath(string baseDirectory, string location)
...
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    CreateNoWindow = true
+                };
+                var process = Process.Start(startInfo);
+
+                process.WaitForExit();
+
+                if (0 != process.ExitCode)
+                    errorMessage = $"invalid conan executable {exe}: conan --version failed with error {process.ExitCode}";
```
Comment (jgsogo) [inline Conan.VisualStudio.Core/ConanPathHelper.cs]:
```suggestion
                    errorMessage = $"Invalid conan executable {exe}: conan --version failed with error {process.ExitCode}";
```

File: Conan.VisualStudio/ConanOptionsPage.cs [hunk 3]
```diff
@@ -14,13 +16,25 @@ public class ConanOptionsPage : DialogPage
         private ConanBuildType? _conanBuild;
         private bool? _conanUpdate;
 
+
+        private bool ValidateConanExecutableAndShowMessage(string exe)
+        {
+            if (!ConanPathHelper.ValidateConanExecutable(exe, out string errorMessage))
+            {
+                MessageBox.Show(errorMessage, "invalid conan executable",
```
Comment (jgsogo) [inline Conan.VisualStudio/ConanOptionsPage.cs]:
```suggestion
                MessageBox.Show(errorMessage, "Invalid conan executable",
```

Uppercase first letter, I think it is nicer


image

**Comment by SSE4** [comment]
that's totally weird, it means changes applied even if we hit cancel? I believe setter should be never called in case of cancelation.

**Pull request merged**